### DISCUSSION TITLE
Switch Colorscheme

### DISCUSSION THREAD

**Discussion by MunifTanjim** [MunifTanjim/nougat.nvim#76] [Q&A] [answered]
hi @MunifTanjim, how can I retrieve RGB color  from a syntax highlighting group? - because I am using multiple colorschemes, and switch it inside nvim, so I want statusline to be compatible with the colorscheme.

_Originally posted by @linrongbin16 in https://github.com/MunifTanjim/nougat.nvim/issues/5#issuecomment-1872463265_

**Comment by MunifTanjim** [comment, marked as answer]
It is mostly documented here: https://github.com/MunifTanjim/nougat.nvim/tree/main/lua/nougat/color

First of all, use `nougat.color` module for picking color for different items:

```lua
local color = require('nougat.color').get()
```

```
color.red
color.accent.red
color.green
color.accent.green
color.yellow
color.accent.yellow
color.blue
color.accent.blue
color.magenta
color.accent.magenta
color.cyan
color.accent.cyan

color.bg
color.accent.bg
color.bg0
color.bg1
color.bg2
color.bg3
color.bg4

color.fg
color.accent.fg
color.fg0
color.fg1
color.fg2
color.fg3
color.fg4
```

If the colorscheme you're using has `nougat.color.<colorscheme>` module, where `<colorscheme>` is the name (e.g. `gruvbox`), `nougat.nvim` will read that. Or if the colorscheme defines `g:terminal_color_*` variables, `nougat.nvim` will automatically extract color from those.

If your colorscheme does neither of those, you can create the `nougat.color.<colorscheme>` module yourself with the following structure:

```lua
local mod = {}

function mod.get()
  ---@class nougat.color.name_of_colorscheme: nougat.color
  local color = { accent = {} }
  -- set the values here
  return color
end

return mod
```

With that setup, `nougat.nvim` will always use the colors from the currently active colorscheme. If you change the colorscheme (using `:colorscheme` command), colors for your statusbar/tabline/winbar will automatically be updated.

---

@linrongbin16 hope that helps. Let me know if you have any other questions.

**Reply by linrongbin16** [reply]
thanks for detailed explain!
